src/chatbots/VaChatbot.ts

```diff
@@ -0,0 +1,41 @@
+import { IAIClient } from '@redhat-cloud-services/ai-client-common';
+import { AsyncStateManager } from '../asyncClientInit/types';
+import { useMemo } from 'react';
+import VAClient from '../aiClients/vaClient';
+import { createClientStateManager } from '@redhat-cloud-services/ai-client-state';
+import VAMessageEntry from '../Components/VAClient/VAMessageEntry';
+import { AsyncStateManagerConfiguration } from '../aiClients/types';
+
+const VaChatbot: AsyncStateManager<IAIClient> = {
+  useStateManager: (): AsyncStateManagerConfiguration<VAClient> => {
+    return useMemo(() => {
+      const client = new VAClient();
+      const stateManager = createClientStateManager(client);
+      return {
+        stateManager,
+        historyManagement: false,
+        docsUrl:
+          'https://docs.redhat.com/en/documentation/red_hat_hybrid_cloud_console/1-latest/html/getting_started_with_the_red_hat_hybrid_cloud_console/hcc-help-options_getting-started#virtual-assistant_getting-started',
+        streamMessages: false,
+        modelName: 'Virtual Assistant',
+        selectionTitle: 'Hybrid Cloud Core console',
+        selectionDescription:
+          'Update your personal information, request access from your admin, show critical vulnerabilities, get Advisor recommendations, and more.',
+        MessageEntryComponent: VAMessageEntry,
+      };
+    }, []);
+  },
+
+  useIsAuthenticated: () => {
```

Comment:
Yup, async hooks aren't working yet. I may try to figure it out with module federation.

Reply:
I mean, adding full support for it to Scalprum. I know the pattern you used above. If we want to use it here, which we do, I'd like to have a generic pattern for all tenants and avoid custom implementation in individual modules. The POC is here: https://github.com/Hyperkid123/async-hooks-pc. I will move on to adding it to Scalprum. There is an additional thing still being explored involving an unknown number of hooks that would be required here. I do not want devs to ever create hook handling components on their own.